test/common/run-tests

```diff
@@ -75,7 +75,7 @@ def print_test(test, print_tap=True, retry_reason=""):
     print()  # Tap needs to start on a separate line
     if test.process.returncode == 0:
         print("ok " + test_name(test) + retry_reason)
-    elif test.process.returncode == 77 or b"# SKIP " in test.output:
+    elif test.process.returncode == 77 or b"# SKIP " in test.output.splitlines()[-1]:
```

Comment:
I don't like the approach.  There's no guarantee in TAP that #skip will always come on the last line.

Why not rather check for something like a line looking like "^ok \d+ # skip" plus/minus noise/case?  That should always be a skip (since other messages should get `#` prefixed)

Reply:
>  There's no guarantee in TAP that #skip will always come on the last line.

Isn't there? Then thats bummer

> Why not rather check for something like a line looking like "^ok \d+ # skip" plus/minus noise/case? 

That `ok \d+...` is added by `test/common/run-tests` and does not come from tap. We need to recognize it in tap and then this script adds that message (see `print_test()`)

Reply:
Hmm, actually it seems that it is guaranteed that #SKIP is on the last line, see lines 79 and 82.

Reply:
I think this is actually the wrong place to look.  This script isn't interpreting its own output.  It's interpreting the output from testlib.  For that, the relevant line is 1578:

```python3
        # Return 77 if all tests were skipped
        if len(skips) == test_count:
            sys.stdout.write("# SKIP {0}\n".format(", ".join(["{0} {1}".format(str(s[0]), s[1]) for s in skips])))
            return 77
        if failures:
            sys.stdout.write("# {0} TEST{1} FAILED {2}\n".format(failures, "S" if failures > 1 else "", details))
            return 1
        else:
            sys.stdout.write("# {0} TEST{1} PASSED {2}\n".format(test_count, "S" if test_count > 1 else "", details))
            return 0
```

and indeed, the last line will have `# SKIP`.  But then that means that the only place we are expecting to see `# SKIP` will also always return 77.  So, do we need this extra check at all, actually?

Reply:
Also `bots/tests-policy` can add `# SKIP` and then returncode is not 77.
Also `bots/tests-policy` can add `# RETRY`.

I think that if we teach test-policy to return 77 and 1 we can simplify this.

Reply:
I __think__ that this logic is triple complicated because tests-policy works also with `not ok ...` part which complicates parsing. If we change it, we will surely break tests as every project would need to use new `test/common/run-tests` immediately. But if we would create copy of tests-policy then we can switch easily and once none is using current tests-policy we can just drop it.